client/Reviews.jsx

```diff
@@ -0,0 +1,235 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import Axios from 'axios';
+import ReviewList from './ReviewList.jsx';
+import Search from './Search.jsx';
+import Ratings from './Ratings.jsx';
+import styles from './styles/AppStyle.css';
+
+class Reviews extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.nextPageClick = this.nextPageClick.bind(this);
+    this.previousPageClick = this.previousPageClick.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+    this.handleChange = this.handleChange.bind(this);
+    this.backToAllReviews = this.backToAllReviews.bind(this);
+
+    this.state = {
+      data: [],
+      copyData: [],
+      count: 0,
+      pageReviews: [],
+      search: '',
+      searchActive: false
+    };
+  }
+
+  componentDidMount() {
+    this.getData();
+  };
+
+  getData() {
+    var parts = document.URL.split("/");
+    var lastSegment = parts.pop() || parts.pop();
+
+    Axios.get(`http://localhost:3003/${lastSegment}`)
+      .then(response => {
+        console.log('this is data ', response.data)
+        this.setState({
+          data: response.data,
+          copyData: response.data,
+          count: response.data.length,
+          pageReviews: response.data.slice(0, 7)
+        })
+      })
+      .catch(err => { // did not handle possible errors before
+        console.error("error in setting state")
+      });
+  };
+
+  allAverage(array) {
+    let count = 0;
+    let divide = array.length * 6;
+
+    array.map(obj => {
+      for (var key in obj.ratings) {
+        count += obj.ratings[key];
+      }
+    })
+
+    return count / divide;
+  };
+
+  nextPageClick() {
+    let list = this.state.pageReviews;
+    let length = list.length - 1;
+    let lastReview = list[length];
+    let lastReviewNum = lastReview.reviewNum;
+    let start;
+    let result = [];
+    if (list.length === 7) { // prevents nextPageClick from rendering a blank page if at the end of total list of reviews
+      for (var i = 0; i < this.state.data.length; i++) {
```

Comment:
instead of looping through all reviews to find the matching lastreview, is there way to skip ahead in the data array as opposed to starting at 0 every time? could help with optimizing if it is possible

Reply:
I'll definitely take a look into this for optimizing, thats a great idea!